bin/smarthome.py

```diff
@@ -318,27 +332,28 @@ def checkConfigFiles(self):
         """
         
         # logging
+        log_default = os.path.join(self._base_dir, 'etc', 'logging' + YAML_FILE + DEFAULT_FILE)
         if not (os.path.isfile(self._log_config)):
-            if os.path.isfile(self._log_config + DEFAULT_FILE):
-                shutil.copy2(self._log_config + DEFAULT_FILE, self._log_config)
+            if os.path.isfile(log_default):
+                shutil.copy2(log_default, self._log_config)
         # smarthome.yaml
+        smarthome_default = os.path.join(self._base_dir, 'etc', 'smarthome' + YAML_FILE + DEFAULT_FILE)
         if not (os.path.isfile(self._smarthome_conf_basename + YAML_FILE)) and not (
                 os.path.isfile(self._smarthome_conf_basename + CONF_FILE)):
-            if os.path.isfile(self._smarthome_conf_basename + YAML_FILE + DEFAULT_FILE):
-                shutil.copy2(self._smarthome_conf_basename + YAML_FILE + DEFAULT_FILE,
-                             self._smarthome_conf_basename + YAML_FILE)
+            if os.path.isfile(smarthome_default):
+                shutil.copy2(smarthome_default, self._smarthome_conf_basename + YAML_FILE)
         # loadable modules
+        module_default = os.path.join(self._base_dir, 'etc', 'module' + YAML_FILE + DEFAULT_FILE)
         if not (os.path.isfile(self._module_conf_basename + YAML_FILE)) and not (
                 os.path.isfile(self._module_conf_basename + CONF_FILE)):
-            if os.path.isfile(self._module_conf_basename + YAML_FILE + DEFAULT_FILE):
-                shutil.copy2(self._module_conf_basename + YAML_FILE + DEFAULT_FILE,
-                             self._module_conf_basename + YAML_FILE)
+            if os.path.isfile(module_default):
+                shutil.copy2(module_default, self._module_conf_basename + YAML_FILE)
         # plugins
+        plugin_default = os.path.join(self._base_dir, 'etc', 'plugin' + YAML_FILE + DEFAULT_FILE)
         if not (os.path.isfile(self._plugin_conf_basename + YAML_FILE)) and not (
                 os.path.isfile(self._plugin_conf_basename + CONF_FILE)):
-            if os.path.isfile(self._plugin_conf_basename + YAML_FILE + DEFAULT_FILE):
-                shutil.copy2(self._plugin_conf_basename + YAML_FILE + DEFAULT_FILE,
-                             self._plugin_conf_basename + YAML_FILE)
+            if os.path.isfile(plugin_default):
+                shutil.copy2(plugin_default, self._plugin_conf_basename + YAML_FILE)
```

Comment:
Wow, a lot of duplicate code. But not introduced by you. Could be reduced by iterating over a list of config files/paths.

Reply:
Agree, but there is inconsistency: `self._log_config` already contain `YAML_FILE` suffix, other configs not. However it shouldn't be hard to fix.